config.json

```diff
@@ -392,6 +392,18 @@
         "control_flow_conditionals",
         "text_formatting"
       ]
+    },
+    {
+      "slug": "affine-cipher",
+      "uuid": "9bc3f040-9e4b-4ed0-b23c-3ae565e83a59",
+      "core": false,
+      "difficulty": 5,
```

Comment:
Do you consider this substantially more difficult than `crypto-square` or `tournament`? Those are `4`s, so we should have an argument here to introduce `5`.

Reply:
PS: I use https://jonmcalder.shinyapps.io/exercism-config-viz/ BTW to visualise where a new exercise might fit into the track structure. Please feel free to suggest an `unlocked_by`.

Reply:
@katrinleinweber (For me) I do not necessarily feel that affine-cipher is difficult at all (compared to crypto-square)(tournament is more difficult). However, I do feel that the reader contends with a high overhead of needing to find out gcd and MMI as pre-requisites of solving the question (I would . As expected I checked in python and crypto-square was 1 whereas tournament and affine-cipher was 5. The rust track does not have affine-cipher but rates tournament and crypto-square both as 4. I am happy to rate affine-cipher at 4, however, to the non-mathematically inclined will experience it to be a  bit harder than normal.

Reply:
ideally. for the student. We would want to break it up into smaller pieces and make each piece as pre-requisite. These are the `unlocked_by` tree; in brackets it is difficulty. 

1.  Ideal Option

```
prime-factors (3)-> gcd (3) -> mmi (2) --> affine-cipher(3)
crypto-square (4) -> affine-cipher(3) 
```
If these pre-requisites exist for affine-cipher, it's difficulty should dampen. And the true task of affine cipher is just to put gcd and mmi together and also handle the normalisation-- that's all. 

2. Practical Option

```
prime-factors (3) ->affine-cipher(4)
```
In addition, I should add the pre-requisite of being able to solve a gcd function in the README.md. I noticed I have not made it as clear as how I have done for the MMI.

The reason I include the practical option is because. 1) `unlocked_by` cannot have a many-to-one dependence (correct me if I am wrong). 2) Do not want to add too many new exercises just for the sake of affine-cipher.